src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/llm_inputs.py

```diff
@@ -206,6 +196,125 @@ def create_llm_inputs(
 
         return json_in_pa_format
 
+    @classmethod
+    def get_generic_dataset_json(
+        cls,
+        input_type: PromptSource,
+        output_format: OutputFormat,
+        dataset_name: str,
+        starting_index: int,
+        length: int,
+        tokenizer: Tokenizer,
+        prompt_tokens_mean: int,
+        prompt_tokens_stddev: int,
+        num_of_output_prompts: int,
+        batch_size: int,
+        input_filename: Optional[Path],
+    ) -> Dict:
+        """
+        Retrieve and convert the dataset based on the input type.
+
+        Parameters
+        ----------
+        input_type:
+            Specify how the input is received
+        output_format:
+            Specify the output format
+        dataset_name:
+            The name of the dataset
+        starting_index:
+            Offset from within the list to start gathering inputs
+        length:
+            Number of entries to gather
+        tokenizer:
+            The tokenizer to use when generating synthetic prompts
+        prompt_tokens_mean:
+            The mean length of the prompt to generate
+        prompt_tokens_stddev:
+            The standard deviation of the length of the prompt to generate
+        num_of_output_prompts:
+            The number of synthetic output prompts to generate
+        batch_size:
+            The number of inputs per request (currently only used for v1/embeddings)
+        input_filename:
+            The path to the input file containing the prompts in JSONL format.
+        Returns
+        -------
+        Dict:
+            The generic dataset JSON
+        """
+        if output_format == OutputFormat.OPENAI_EMBEDDINGS:
```

Comment:
This is definitely unclean at this point and we need to refactor this file.
We will add this later, so not blocking this PR. Just want to document that this is was expected and we need to refactor later.

Reply:
Great point, thanks for calling this out. I agree that we'll need to update this. It might simply be breaking it out into functions once rankings are added (though then we'll just have more of these passthrough functions that create a deep stack that passes all the args through). Or there might be a better way once we finalize the service kind or endpoint changes (e.g. in Neelay's tickets).